aiidalab_widgets_base/nodes.py

```diff
@@ -210,22 +215,29 @@ def _find_called(cls, root):
     @classmethod
     def _find_outputs(cls, root):
         process_node = load_node(root.parent_pk)
-        if root.namespace:
-            outputs = process_node.outputs[root.namespace]
-        else:
-            outputs = process_node.outputs
 
+        # Gather outputs from node and its namespaces (if present):
+        outputs = functools.reduce(
+            lambda d, namespace: d[namespace],
+            root.namespaces or [],
+            process_node.outputs,
+        )
```

Comment:
No worries, I admit this is kind of confused, every time I look at it again I need to go through all the code and try to clean my mind.

Assuming we have the following outputs of a process node `wc_node`:
```
ns00
    ns10
        outnode_a
        outnode_b
ns01
        outnode_a
        outnode_c
```
If we want to access `outnode_a` node, it should be `wc_node.outputs['ns00']['ns01']['outnode_a']`. In the scope (iteration), the `root.namespaces` in this layer equal to `[`ns00`, `ns01`]`.

> Either way, this function probably requires a bit more commenting since it is somewhat intransparent.
Indeed, I will add it if you think the code and explanation above are clear enough.

Reply:
`d[namespace]` in the `reduce` make sure that the `d` accumulate the change for every nested `namespace`.

Reply:
From offline discussion: The code appears to be correct, however it will require a bit more documentation to make it easier to follow. After that I'm happy to approve and merge.

Reply:
Hi @csadorf, I add docstring and comment to make (try to🤣) it more clear of the logic. Please have a look.